components/clarinet-format/src/formatter/mod.rs

```diff
@@ -1577,14 +1637,30 @@ mod tests_formatter {
         let result = format_with_default(src);
         assert_eq!(src, result);
     }
+
+    #[test]
+    fn define_multiline_list() {
+        let src = r#"  (
+    (optional <sip-010>) ;; token
+    uint                 ;; amount
+    principal            ;; with
+    uint                 ;; nonce
```

Comment:
I know this is my example code, but I will reiterate that I would also be okay with the formatter rejecting these trailing comments and forcing a pre-comment, e.g.:

```clarity
  (
    ;;token
    (optional <sip-010>)
    ;; amount
    uint
    ;; with
    principal
    ;; nonce
    uint
   )
```

Reply:
I think it's fine to have the aligned eol comments. It wasn't the main bug here anyway. The fix for this was properly handling multi-line lists. The comments are just a cherry on top. Besides, the formatter goes through in order, so if we made them into pre-comments, we'd have to re-order the PSE, or do a read from the end.. which would get messy